docs/src/lang/option-types.md

```diff
@@ -0,0 +1,386 @@
+# Option Types
+
+[[Back to Apalache extensions]](./apalache-extensions.md)
+
+[Option types](https://en.wikipedia.org/wiki/Option_type) are useful when you
+want to internalize reasoning about partial functions. A simple motivating
+example is division over integers, for which `n/0` is undefined.
+
+The basic idea is as follows: given a partial function `f : A -> B`, we form the
+type `Option(B)` by extending `B` with an element representing a missing value,
+`None`, and lift each value `b` in `B` to `Some(b)`, allowing us to represent
+the partial function `pf : A -> Option(B)`, such that, for each `a` in `A`,
+`pf(a) = Some(f(a))` iff `f(a)` is defined, and `None` otherwise.
+
+
+Apalache leverages its support for [variants](./variants.md) to define a
+polymorphic option type along with some common utility functions in the module
+[Option.tla][]. 
+
+The module defines a type alias `$option` as
+
+```tla
+\* @typeAlias: option = Some(a) | None(UNIT);
+```
+
+However, due to the current lack of support for polymorphic aliases, this alias
+has limited utility, and  parametric option types can only be properly expressed
+with the variant type `Some(a) | None(UNIT)`. Nonetheless, in this manual
```

Comment:
I first thought the variant differs from the definition above. Maybe better to explain:

```suggestion
has limited utility, and parametric option types can only be properly expressed
by writing out the full variant type `Some(a) | None(UNIT)`. Nonetheless, in this manual
```